bindings/java/java/src/main/java/org/ethereum/evmc/Host.java

```diff
@@ -77,7 +77,7 @@ static ByteBuffer copy_code(int context_index, byte[] address, int code_offset)
             "HostContext does not exist for context_index: " + context_index);
     byte[] code = context.getCode(address).array();
 
-    if (code != null && code_offset > 0 && code_offset < code.length) {
```

Comment:
Is this change still relevant? It seems to have been missed in #535.

Reply:
Right - I can't remember if that change is required. I left it out. It doesn't hurt to check for those.

Reply:
OK I remember now. The code can never be null, since it's a byte array extracted from a ByteBuffer object. The `code_offset > 0` should be removed imo since none of the other host impls check for that.

Reply:
`code_offset` can be 0.

Reply:
Actually this entire `copy_code_fn` and it's Java counterpart is broken. It never copies anything back, that's also why the FIXME for `buffer_size` is there.

I suggest this is fixed independently.

Reply:
I suggest this is left to be done in #545.